### ISSUE TITLE
List of actions for code formating

### ISSUE THREAD

**Issue by ye-luo** [QMCPACK/qmcpack#1339]
We are aiming at a full repo (src) reformatting very soon.
If you have significant changes not merged yet, please comment.
We are making a plan and strategy to minimize impact on every developer.

To be completed before the action
1. [x] Update manual with day-to-day formatting procedure (@PDoakORNL )
2. [x] Merge PR #879 (msdj orbital optimization)
3. [x] Merge PR #1279 (GPU delayed updates)
4. [x] Merge CUDA AFQMC code or avoid initially reformatting AFQMC
5. [x] Pick a date for reformatting.
5. [x] Reformat.
6. [x] Discuss with and help authors of any pending PRs at time of reformat

**Comment by prckent** [comment]
For anyone new to this topic, see discussion in #794

**Comment by prckent** [comment]
Please do not unpin this issue.

**Comment by PDoakORNL** [comment]
For the first checkbox I propose adding an href from the manual to the appropriate section of the wiki.  At any particular time the best way we've found to do the formatting and its current place in the workflow should be there.

**Comment by PDoakORNL** [comment]
I propose creating the reformatting PR tomorrow.

**Comment by ye-luo** [comment]
Can we start with a trial PR formatting one or two folders under src now to see if there is any issue?

**Comment by PDoakORNL** [comment]
What we are talking about with function calls clang-format call bin-packing.
It's not very flexible i.e. it either goes to the max width or does one per line.
I prefer one per line to parsing several lines of max character width arguments but not enough to be a no vote.

Since @ye-luo is a strong BinPackArguments:true vote it's currently 1:0 to change this.

Thumbs up to change to true for remainder of reformat and future use. 
Voting will when the next reformat PR is merged and currently it will be changed to true.

```C++
BinPackArguments (bool)
```
If false, a function call’s arguments will either be all on the same line or will have one line each.
```C++
    true:
    void f() {
      f(aaaaaaaaaaaaaaaaaaaa, aaaaaaaaaaaaaaaaaaaa,
        aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa);
    }

    false:
    void f() {
      f(aaaaaaaaaaaaaaaaaaaa,
        aaaaaaaaaaaaaaaaaaaa,
        aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa);
    }
```

**Comment by ye-luo** [comment]
@PDoakORNL I'd like to have BinPackArguments true but BinPackParameters false. Do you have any preference of BinPackParameters?

**Comment by ye-luo** [comment]
from `src/QMCWaveFunctions/BsplineFactory/SplineC2RAdoptor.h` with `BinPackArguments=false`
```
      const ST f3_xxx_r = t3_contract(gh000[jr],
                                      gh001[jr],
                                      gh002[jr],
                                      gh011[jr],
                                      gh012[jr],
                                      gh022[jr],
                                      gh111[jr],
                                      gh112[jr],
                                      gh122[jr],
                                      gh222[jr],
                                      g00,
                                      g01,
                                      g02,
                                      g00,
                                      g01,
                                      g02,
                                      g00,
                                      g01,
                                      g02);
```

**Comment by PDoakORNL** [comment]
BinPackParameters will remain false I think we have broad agreement on that.

As far as the previously done directories, we have to format those directories again.  I was thinking I'd do the entire source tree once the incrementtals are done.  That way we will also pick up files that are added or deformatted during the formatting PR.

**Comment by prckent** [comment]
Closed by #1583
